src/backend/cdb/motion/cdbmotion.c

```diff
@@ -1274,3 +1280,115 @@ UpdateSentRecordCache(int32 *sent_record_typmod)
 		*sent_record_typmod = NextRecordTypmod;
 	}
 }
+
+static int
+GetICTypeByName(const char *type_name)
+{
+	for (int i = 0; i < CurrentIPCLayerImplNum; ++i)
+	{
+		if (!pg_strcasecmp(IPCLayerImpls[i]->type_name, type_name))
+			return IPCLayerImpls[i]->ic_type;
+	}
+
+	return -1;
+}
+
+/*
+ * try best to find the most appropriate IPC layer implement.
+ */
+static int
+TryFindICType()
+{
+	/* IPCLayerImpls[0]->ic_type is valid at least */
+	Assert(CurrentIPCLayerImplNum);
+
+	int candidate_types_order[IPCLAYER_CANDIDATE_NUM] =
```

Comment:
The array size could be omitted, so no need to define an additional macro that must be consistent with the length of the array content.

Reply:
fix with [commit](https://github.com/cloudberrydb/cloudberrydb/commit/8e8c769beb5dcf4fe8456ea299350620201e5275)